pkg/proxy/util/iptables/traffic.go

```diff
@@ -91,3 +91,33 @@ func (d *detectLocalByCIDR) JumpIfNotLocal(args []string, toChain string) []stri
 	klog.V(4).Info("[DetectLocalByCIDR (", d.cidr, ")]", " Jump Not Local: ", line)
 	return line
 }
+
+type detectLocalByInterface struct {
+	ifaceName string
+}
+
+// NewDetectLocalByInterface implements the LocalTrafficDetector interface using an interface or a bridge.
+// This can be used when a pod interface name/prefix can be used to capture the notion of local traffic.
+func NewDetectLocalByInterface(ifaceName string) (LocalTrafficDetector, error) {
+	maxIfNameSize := 16 // linux/if.h
+	if len(ifaceName)+1 > maxIfNameSize || ifaceName == "" {
+		return nil, fmt.Errorf("invalid interface name")
+	}
```

Comment:
This should be done here
https://github.com/kubernetes/kubernetes/blob/ee297b6f4d78daaf2f71c78a05203698581ca40a/pkg/proxy/apis/config/validation/validation.go#L39
and add tests in validation_test.go

I personally think we should validate that the prefix len is >= 4 (i.e "veth") or <= than (max-7) (ie. 2^7 is 128) , preventing people foot gun

@danwinship @thockin , what do you think?

Reply:
@aojea : yeah it makes sense, I guess in plugins like sdn that use kube-proxy we use the function directly and don't use the validation parts via api, so I added it here. Makes sense to move the validation separately into validation.go.

Reply:
quick comment about the minimum length validation: Azure CNI uses the prefix "azv", so if you decide to add a minimum length requirement could you set it >= 3 (or lower?)

Reply:
yeah sure, I will talk with Antonio about this. For now I didn't add the min length part because like you are pointing out different platforms could be using different naming schemes, and in linux kernel `IFNAMSIZ` only includes an upper bound. If folks really want to put a lower bound which is not 0, maybe 2 ?